http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -238,18 +222,17 @@ class LeoPubsubMessageSubscriber[F[_]](
       .parEvalMapUnordered(config.concurrency)(messageHandler)
       .handleErrorWith(error => Stream.eval(logger.error(error)("Failed to initialize message processor")))
 
-  private def ack(event: CloudPubsubEvent[LeoPubsubMessage]): F[Unit] =
+  private def ack(event: ReceivedMessage[LeoPubsubMessage]): F[Unit] =
     for {
       _ <- logger.info(s"acking message: ${event}")
-      _ <- event match {
-        case CloudPubsubEvent.GCP(event) =>
-          F.delay(
-            event.consumer.ack()
-          )
-        case CloudPubsubEvent.Azure(_) => F.unit
-      }
+
+      _ <- F.delay(event.ackHandler.ack())
+
       end <- F.realTimeInstant
-      duration = (end.toEpochMilli - event.publishedTime.toEpochMilli).millis
+      // A zero duration means the publishing time is not available, this should never be case
+      // and it must be debugged if it happens.
+      duration = Duration.create( end.toEpochMilli - event.publishedTime.getOrElse(Instant.now()).toEpochMilli,
```

Comment:
event.publishedTime.getOrElse(end) is more correct...the `Instant.now()` may have drifted a bit from `end`